From: Qore Technologies <info@qoretechnologies.com>
Date: Fri, 2 Oct 2026 00:00:00 +0000
Subject: Correct grpcio-tools build metadata and language-specific declarations

Keep Python 3 semantics explicit for Cython, declare the wheel license with
SPDX, remove the nonexistent manifest entry, and restrict RTTI flags to C++.
Three bundled Protobuf always_inline definitions need the inline declaration
required by GCC; their bodies and call sites remain unchanged.

Copyright 2026 Qore Technologies, s.r.o.
License: Apache-2.0 (gRPC); BSD-3-Clause (Protobuf)

--- a/setup.py
+++ b/setup.py
@@ -56,7 +56,6 @@
     "Development Status :: 5 - Production/Stable",
     "Programming Language :: Python",
     "Programming Language :: Python :: 3",
-    "License :: OSI Approved :: Apache Software License",
 ]
 
 PY3 = sys.version_info.major == 3
@@ -144,7 +143,8 @@
         def new_compile(obj, src, ext, cc_args, extra_postargs, pp_opts):
             if src.endswith(".c"):
                 extra_postargs = [
-                    arg for arg in extra_postargs if "-std=c++" not in arg
+                    arg for arg in extra_postargs
+                    if "-std=c++" not in arg and arg not in ("-frtti", "-fno-rtti")
                 ]
             elif src.endswith(".cc") or src.endswith(".cpp"):
                 extra_postargs = [
@@ -308,7 +308,7 @@
     if BUILD_WITH_CYTHON:
         from Cython import Build
 
-        return Build.cythonize(extensions)
+        return Build.cythonize(extensions, compiler_directives={"language_level": 3})
     else:
         return extensions
 
@@ -326,7 +326,7 @@
         "Source Code": "https://github.com/grpc/grpc/tree/master/tools/distrib/python/grpcio_tools",
         "Bug Tracker": "https://github.com/grpc/grpc/issues",
     },
-    license="Apache License 2.0",
+    license="Apache-2.0 AND BSD-3-Clause AND MIT",
     classifiers=CLASSIFIERS,
     ext_modules=extension_modules(),
     packages=setuptools.find_packages("."),
--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -2,7 +2,6 @@
 include _spawn_patch.py
 include grpc_version.py
 include python_version.py
-include protoc_deps.py
 include protoc_lib_deps.py
 include README.rst
 include grpc_tools/grpc_version.py
--- a/third_party/protobuf/src/google/protobuf/generated_message_tctable_lite.cc
+++ b/third_party/protobuf/src/google/protobuf/generated_message_tctable_lite.cc
@@ -714,7 +714,7 @@
 
 // Prefetch the enum data, if necessary.
 // We can issue the prefetch before we start parsing the ints.
-PROTOBUF_ALWAYS_INLINE void PrefetchEnumData(uint16_t xform_val,
+PROTOBUF_ALWAYS_INLINE inline void PrefetchEnumData(uint16_t xform_val,
                                              TcParseTableBase::FieldAux aux) {
 }
 
@@ -726,7 +726,7 @@
 // way more common than the kTvEnum cases. It is also called from places that
 // already have out-of-line functions (like MpVarint) so an extra out-of-line
 // call to `ValidateEnum` does not affect much.
-PROTOBUF_ALWAYS_INLINE bool EnumIsValidAux(int32_t val, uint16_t xform_val,
+PROTOBUF_ALWAYS_INLINE inline bool EnumIsValidAux(int32_t val, uint16_t xform_val,
                                            TcParseTableBase::FieldAux aux) {
   if (xform_val == field_layout::kTvRange) {
     auto lo = aux.enum_range.start;
@@ -803,7 +803,7 @@
 }
 
 template <typename FieldType>
-PROTOBUF_ALWAYS_INLINE const char* TcParser::FastVarintS1(
+PROTOBUF_ALWAYS_INLINE inline const char* TcParser::FastVarintS1(
     PROTOBUF_TC_PARAM_DECL) {
   using TagType = uint8_t;
   if (PROTOBUF_PREDICT_FALSE(data.coded_tag<TagType>() != 0)) {
